runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java

```diff
@@ -594,28 +601,41 @@ private void setTask(String name, int amt)
 
 	private void addCounter()
 	{
-		if (!config.showInfobox() || counter != null || Strings.isNullOrEmpty(taskName))
+		if (config.showInfobox() && counter == null && !Strings.isNullOrEmpty(taskName))
 		{
-			return;
-		}
+			Task task = Task.getTask(taskName);
+			int itemSpriteId = ItemID.ENCHANTED_GEM;
+			if (task != null)
+			{
+				itemSpriteId = task.getItemSpriteId();
+			}
 
-		Task task = Task.getTask(taskName);
-		int itemSpriteId = ItemID.ENCHANTED_GEM;
-		if (task != null)
-		{
-			itemSpriteId = task.getItemSpriteId();
+			BufferedImage taskImg = itemManager.getImage(itemSpriteId);
+			counter = new TaskCounter(taskImg, this, amount);
+			infoBoxManager.addInfoBox(counter);
 		}
 
-		BufferedImage taskImg = itemManager.getImage(itemSpriteId);
-		final String taskTooltip = ColorUtil.prependColorTag("%s</br>", new Color(255, 119, 0))
-			+ ColorUtil.wrapWithColorTag("Pts:", Color.YELLOW)
-			+ " %s</br>"
-			+ ColorUtil.wrapWithColorTag("Streak:", Color.YELLOW)
-			+ " %s";
-		counter = new TaskCounter(taskImg, this, amount);
-		counter.setTooltip(String.format(taskTooltip, capsString(taskName), points, streak));
+		if (counter != null && !Strings.isNullOrEmpty(taskName))
+		{
+			String taskTooltip = ColorUtil.prependColorTag("%s</br>", new Color(255, 119, 0))
+					+ ColorUtil.wrapWithColorTag("Pts:", Color.YELLOW)
+					+ " %s</br>"
+					+ ColorUtil.wrapWithColorTag("Streak:", Color.YELLOW)
+					+ " %s";
 
-		infoBoxManager.addInfoBox(counter);
+			// makes it so upon updating to track initialAmount people's previously active task won't show X/-1
+			if (initialAmount != -1)
+			{
+				taskTooltip += "</br>"
```

Comment:
you can just append the initialAmount here instead of using String.format for it and then you can remove the counter.setTooltip duplicated code